src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java

```diff
@@ -832,6 +836,213 @@ public void testNoSpaceInsert() throws Exception {
         }
     }
 
+    /**
+     * Test bulk insert with all temporal types and money as varchar when useBulkCopyForBatchInsert is true.
+     * sendTemporalDataTypesAsStringForBulkCopy is set to true by default.
+     * Temporal types are sent as varchar, and money/smallMoney are sent as their respective types.
+     * 
+     * @throws Exception
+     */
+    @Test
+    @Tag(Constants.xAzureSQLDW)
+    public void testBulkInsertWithAllTemporalTypesAndMoneyAsVarchar() throws Exception {
+        String tableName = "BulkTable";
```

Comment:
Use unique object names to avoid concurrent test runs from stepping on each other:
```suggestion
        String tableName = RandomUtil.getIdentifier("BulkTable");
```